new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonUtils.java

```diff
@@ -0,0 +1,655 @@
+
+    // create an ingress in domain namespace
+    String ingressName = domainUid + "-nginx";
+    List<String> ingressHostList =
+        createIngress(ingressName, domainNamespace, domainUid, managedServerPort, clusterNames);
+
+    assertNotNull(ingressHostList,
+        String.format("Ingress creation failed for domain %s in namespace %s", domainUid, domainNamespace));
+
+    // check the ingress was found in the domain namespace
+    assertThat(assertDoesNotThrow(() -> listIngresses(domainNamespace)))
+        .as("Test ingress {0} was found in namespace {1}", ingressName, domainNamespace)
+        .withFailMessage("Ingress {0} was not found in namespace {1}", ingressName, domainNamespace)
+        .contains(ingressName);
+
+    logger.info("ingress {0} for domain {1} was created in namespace {2}",
+        ingressName, domainUid, domainNamespace);
+
+    return ingressHostList;
+  }
+
+  /**
+   * Check pod is created.
+   *
+   * @param podName pod name to check
+   * @param domainUid the domain in which the pod exists
+   * @param domainNamespace the domain namespace in which the domain exists
+   */
+  public static void checkPodCreated(String podName, String domainUid, String domainNamespace) {
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for pod {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                podName,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(assertDoesNotThrow(() -> podExists(podName, domainUid, domainNamespace),
+            String.format("podExists failed with ApiException for pod %s in namespace %s",
+                podName, domainNamespace)));
+  }
+
+  /**
+   * Check pod is ready.
+   *
+   * @param podName pod name to check
+   * @param domainUid the domain in which the pod exists
+   * @param domainNamespace the domain namespace in which the domain exists
+   */
+  public static void checkPodReady(String podName, String domainUid, String domainNamespace) {
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for pod {0} to be ready in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                podName,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(assertDoesNotThrow(() -> podReady(podName, domainUid, domainNamespace),
+            String.format("podReady failed with ApiException for pod %s in namespace %s",
+               podName, domainNamespace)));
+  }
+
+  /**
+   * Check service is created.
+   *
+   * @param serviceName service name to check
+   * @param domainNamespace the domain namespace in which the service exists
```

Comment:
services do not exist in domain namespaces

Reply:
fixed